sigauth/signature.go

```diff
@@ -152,7 +157,7 @@ func (s *SignatureAuth) CheckRequestSignature(r *http.Request, requiredServiceID
 		return "", fmt.Errorf("error building signature string: %v", err)
 	}
 
-	err = s.CheckSignature(sigAuthHeader.KeyId, []byte(sigString), sigAuthHeader.Signature)
+	err = s.CheckServiceSignature(sigAuthHeader.KeyId, []byte(sigString), sigAuthHeader.Signature)
```

Comment:
You'll probably want to rename this function to `CheckRequestServiceSignature` and add a similar `CheckRequestSignature` that takes a public key directly.